tests/integration_tests/datasource_tests.py

```diff
@@ -297,6 +297,44 @@ def test_save(self):
                 print(k)
                 self.assertEqual(resp[k], datasource_post[k])
 
+    def test_save_default_endpoint_validation_fail(self):
+        self.login(username="admin")
+        tbl_id = self.get_table(name="birth_names").id
+
+        datasource_post = get_datasource_post()
+        datasource_post["id"] = tbl_id
+        datasource_post["owners"] = [1]
+        datasource_post["default_endpoint"] = "http://www.google.com"
+        data = dict(data=json.dumps(datasource_post))
+        resp = self.client.post("/datasource/save/", data=data)
+        assert resp.status_code == 500
```

Comment:
Should we really be expecting 500? This may just be me, but when I see 500, I immediately think "the system borked". Could 422 work better here?

Reply:
The frontend side is set to show a error message for 500. Will investigate if it's not too hardwired